libraries/azurerm_network_watcher.rb

```diff
@@ -22,20 +22,27 @@ class AzurermNetworkWatcher < AzurermSingularResource
   ).freeze
 
   attr_reader(*ATTRS)
+  attr_accessor(:nsg)
 
   def initialize(resource_group: nil, name: nil)
     resp = management.network_watcher(resource_group, name)
     return if has_error?(resp)
 
     assign_fields(ATTRS, resp)
 
+    @resource_group = resource_group
     @exists = true
   end
 
   def to_s
     "'#{name}' Network Watcher"
   end
 
+  def flow_logs
+    return nil if @nsg.nil?
```

Comment:
This could be changed to `!!@nsg`. Have seen elsewhere that this preferred over `.nil?`